src/components.jl

```diff
@@ -52,7 +52,7 @@ end
 
 "get an array of tuples of points on a circle with radius `r`"
 function partialcircle(start_θ, end_θ, n = 20, r=1)
-    Tuple{Float64,Float64}[(r*cos(u),r*sin(u)) for u in range(start_θ, stop=end_θ, length=n)]
```

Comment:
The removal of the type annotation here is only necessary for creating circles with a radius `::MyType`. The type in these tuples will be that of `*(::MyType, ::Float64)`, but I don't think annotating that changes the type of the vector. 

If this annotation is very important for performance, this change could easily be reverted, it's not a very important function.